subprojects/plugins/src/main/java/org/gradle/api/plugins/JavaBasePlugin.java

```diff
@@ -139,6 +142,7 @@ private JavaPluginConvention addExtensions(final ProjectInternal project) {
         project.getConvention().getPlugins().put("java", javaConvention);
         project.getExtensions().add(SourceSetContainer.class, "sourceSets", javaConvention.getSourceSets());
         project.getExtensions().create(JavaPluginExtension.class, "java", DefaultJavaPluginExtension.class, javaConvention, project);
+        project.getExtensions().add("javaInstalls", javaInstallationRegistry);
```

Comment:
:x: should specify the public type otherwise the Kotlin DSL generated accessors will use the implementation type which is internal
```suggestion
        project.getExtensions().add(JavaInstallationRegistry.class, "javaInstalls", javaInstallationRegistry);
```

Reply:
I don't have a strong opinion on this, but would it make sense to nest this under the `java` extension?

Reply:
I don't think so. You need a JVM for things other than building things implemented in Java, for example running tools that happen to run on the JVM or for building things implemented in a language other than Java. I'd like to keep slowly moving towards making Java less special.